### ISSUE TITLE
[BUG] Trace: Rendered trace fails to show selected dropdown option

### ISSUE THREAD

**Issue by ydimov-a2z** [microsoft/playwright#12895]
**Context:**
- Playwright Version: v1.20
- Operating System: Mac v12.2.1
- Node.js version: v14.17.6
- Browser: Chromium
- Language: Typescript

## System:
 - OS: macOS 12.2.1
 - Memory: 20.04 MB / 16.00 GB
## Binaries:
 - Node: 14.17.6 - /usr/local/bin/node
 - Yarn: 1.22.17 - /usr/local/bin/yarn
 - npm: 6.14.15 - /usr/local/bin/npm
## Languages:
 - Bash: 3.2.57 - /bin/bash

 **Describe the bug**
My page object has a function to allow selecting an `<option>` from a `<select>` element by its index. When I run the test in `--headed` mode I can see that the correct option is selected and reflected in the UI. However if I view the trace file for the same exact run, the trace will fail to show the option I selected which can cause issues when debugging as it's harder to tell if the test was executing as expected. 

In the example provided below I'm selecting the `SACRAMENTO, SACRAMENTO (IN CITY LIMITS)` option but the rendered trace shows that the `After` screenshot has the placeholder `Please Select An Option` still selected.

<img width="2255" alt="Screen Shot 2022-03-18 at 4 16 59 PM" src="https://user-images.githubusercontent.com/75811951/159093030-95ffac0c-e7c0-4547-bbf4-258b5ea1625a.png">

I'm launching the trace viewer from the built in `html` reporter's report using Chrome browser with latest current version (v99.0.4844.74 at the time I'm writing this)

P.S. I love the framework so far!
 
**Code Snippet**
```typescript
// Page object function:

  /**
   * By default the first city/county option is picked.
   * Note that although the dropdown is 0 indexed the 0 index option is always the "Please Select An Option"
   * @param index
   */
  async selectCityCountyOptionByIndex(index = 1) {
    await this.clickCityCountyDropdown()
    await this.cityCountyDropdown.selectOption({index})
  }

// Sample HTML:

<select class="cell small-6" data-testid="city-county-dropdown" name="cityCounty" style="padding-right: 25px;">
	<option value="">Please Select An Option</option>
	<option value="451594SACRAMENTO, SACRAMENTOtrue">SACRAMENTO, SACRAMENTO (IN CITY LIMITS)</option> 
        <option value="451594SACRAMENTO, SACRAMENTOfalse">SACRAMENTO, SACRAMENTO (NOT IN CITY LIMITS</option
</select>

**Comment by ydimov-a2z** [comment]
Thanks for fixing this @dgozman !

**Comment by NikkTod** [comment]
Hope this fix comes also to the Java playwright

**Comment by mxschmitt** [comment]
> Hope this fix comes also to the Java playwright

Yup! Gets released with 1.21.
